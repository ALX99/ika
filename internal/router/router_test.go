package router

import (
	"testing"

	"github.com/alx99/ika/internal/config"
	"github.com/gkampitakis/go-snaps/snaps"
)

func Test_makeRoutes(t *testing.T) {
	tests := []struct {
		name string // description of this test case
		// Named input parameters for target function.
		rp     string
		nsName string
		route  config.Path
	}{
		{
			name:   "root namespace with no methods",
			rp:     "/test",
			nsName: "root",
			route:  config.Path{},
		},
		{
			name:   "namespaced route with no methods",
			rp:     "/test",
			nsName: "/namespace",
			route:  config.Path{},
		},
		{
			name:   "host route with no methods",
			rp:     "/test",
			nsName: "example.com",
			route:  config.Path{},
		},
		{
			name:   "root namespace with methods",
			rp:     "/test",
			nsName: "root",
			route:  config.Path{Methods: []config.Method{"GET", "POST"}},
		},
		{
			name:   "namespaced route with methods",
			rp:     "/test",
			nsName: "/namespace",
			route:  config.Path{Methods: []config.Method{"GET", "POST"}},
		},
		{
			name:   "host route with methods",
			rp:     "/test",
			nsName: "example.com",
			route:  config.Path{Methods: []config.Method{"GET", "POST"}},
		},
		{
			name:   "host with empty route",
			rp:     "",
			nsName: "example.com",
			route:  config.Path{Methods: []config.Method{"GET", "POST"}},
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := makeRoutes(tt.rp, tt.nsName, tt.route)
			snaps.MatchSnapshot(t, got)
		})
	}
}
