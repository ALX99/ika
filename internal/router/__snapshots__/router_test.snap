
[Test_makeRoutes/root_namespace_with_no_methods - 1]
[]router.routePattern{
    {pattern:"/test", isNamespaced:false},
}
---

[Test_makeRoutes/namespaced_route_with_no_methods - 1]
[]router.routePattern{
    {pattern:"/namespace/test", isNamespaced:true},
}
---

[Test_makeRoutes/host_route_with_no_methods - 1]
[]router.routePattern{
    {pattern:"example.com/test", isNamespaced:false},
}
---

[Test_makeRoutes/root_namespace_with_methods - 1]
[]router.routePattern{
    {pattern:"GET /test", isNamespaced:false},
    {pattern:"POST /test", isNamespaced:false},
}
---

[Test_makeRoutes/namespaced_route_with_methods - 1]
[]router.routePattern{
    {pattern:"GET /namespace/test", isNamespaced:true},
    {pattern:"POST /namespace/test", isNamespaced:true},
}
---

[Test_makeRoutes/host_route_with_methods - 1]
[]router.routePattern{
    {pattern:"GET example.com/test", isNamespaced:false},
    {pattern:"POST example.com/test", isNamespaced:false},
}
---

[Test_makeRoutes/host_with_empty_route - 1]
[]router.routePattern(nil)
---
